go/tasks/plugins/k8s/sagemaker/custom_training.go

```diff
@@ -94,6 +94,26 @@ func (m awsSagemakerPlugin) buildResourceForCustomTrainingJob(
 		Value: strconv.FormatBool(true),
 	})
 
+	if sagemakerTrainingJob.GetTrainingJobResourceConfig() == nil {
+		logger.Errorf(ctx, "TrainingJobResourceConfig is nil")
+		return nil, pluginErrors.Errorf(pluginErrors.BadTaskSpecification, "TrainingJobResourceConfig is nil")
+	}
+
+	if sagemakerTrainingJob.GetTrainingJobResourceConfig().GetDistributedProtocol() == flyteSageMakerIdl.DistributedProtocol_MPI && sagemakerTrainingJob.GetTrainingJobResourceConfig().GetInstanceCount() > 1 {
+		// inject sagemaker_mpi_enabled=true into hyperparameters if the user code designates MPI as its distributed training framework and the instance count is > 1
```

Comment:
what happens when sagemaker_mpi_enabled=true but instance count == 0? is this an error or is the fallback behavior (injecting sagemaker_mpi_enabled=false) intended?

Reply:
@katrogan This particular scenario leads to an error captured by the SageMaker CRD's own validation with the following error message: 
```
spec.resourceConfig.instanceCount: Invalid value: 1: spec.resourceConfig.instanceCount in body should be greater than or equal to 1
```

Reply:
got it. do we want to explicitly pass along this misconfiguration error in this case or is the intention to just fall back to the false case?

Reply:
@katrogan 

That's actually a good catch. I just found a document (https://aws.amazon.com/blogs/machine-learning/multi-gpu-and-distributed-training-using-horovod-in-amazon-sagemaker-pipe-mode/) saying that MPI works on a single instance with multiple GPUs. I also did an experiment and found that MPI can even work on a single instance with one GPU (although that means there's no benefit and even has overhead, it still "works" without an error)


Back to our discussion, so I guess what you are suggesting is we don't do this override on `sagemaker_mpi_enabled` and just let it be carried over to the CRD and let the CRD's validation take care of it?

Reply:
yeah exactly. i would prefer to explicitly report misconfiguration errors to users rather than silently impact behavior and possibly lead to confusion. but i also have very little context with sagemaker in general, so i'm not sure if it's a preferable user experience to mask the error and disable MPI

Reply:
@katrogan Great suggestion. I've fixed it in the following commits. Please take another look